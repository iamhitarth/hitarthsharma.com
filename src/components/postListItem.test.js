import React from 'react'
import { render, cleanup } from '@testing-library/react'
import PostListItem from './postListItem'

describe('Post list item', () => {
  const node = {
    id: '1',
    timeToRead: 3,
    excerpt: 'This is the excerpt',
    frontmatter: {
      title: 'My test post',
    },
    fields: {
      slug: 'my-test-post',
      postDate: '21 October, 2018',
    },
  }

  afterEach(cleanup)

  it('matches snapshot', () => {
    const { container } = render(<PostListItem node={node} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('displays post title', () => {
    const { getByTestId } = render(<PostListItem node={node} />)
    expect(getByTestId('postTitle')).toHaveTextContent(node.frontmatter.title)
  })

  it('displays post date', () => {
    const { getByTestId } = render(<PostListItem node={node} />)
    expect(getByTestId('postDate')).toHaveTextContent(node.fields.postDate)
  })

  it('displays read time', () => {
    const { getByTestId } = render(<PostListItem node={node} />)
    expect(getByTestId('timeToRead')).toHaveTextContent(
      `${node.timeToRead} mins.`
    )
  })

  it('displays post excerpt', () => {
    const { getByTestId } = render(<PostListItem node={node} />)
    expect(getByTestId('postExcerpt')).toHaveTextContent(node.excerpt)
  })
})
