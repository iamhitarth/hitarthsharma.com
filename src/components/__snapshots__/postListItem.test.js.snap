// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post list item matches snapshot 1`] = `
.c2 {
  color: #bbb;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 1.175rem;
}

.c1 {
  margin-top: 1.175rem;
}

.c0 {
  color: inherit;
}

<link
  class="c0"
  to="my-test-post"
>
  <h3
    class="c1"
    data-testid="postTitle"
  >
    My test post
  </h3>
  <h5
    class="c2"
  >
    <span
      data-testid="postDate"
    >
      21 October, 2018
    </span>
    <span
      data-testid="timeToRead"
    >
      3 mins.
    </span>
  </h5>
  <p
    data-testid="postExcerpt"
  >
    This is the excerpt
  </p>
</link>
`;
